avm/res/kubernetes-runtime/load-balancer/tests/e2e/defaults/dependencies.bicep

```diff
@@ -0,0 +1,174 @@
+  name: '${uniqueString(deployment().name, location)}-managedCluster'
+  scope: resourceGroup()
+  params: {
+    name: clusterName
+    kubernetesVersion: '1.32.6'
+    publicNetworkAccess: 'Enabled'
+    azurePolicyEnabled: false
+    disableLocalAccounts: false
+    managedIdentities: {
+      systemAssigned: true
+    }
+    aadProfile: {
+      aadProfileEnableAzureRBAC: true
+      aadProfileManaged: true
+    }
+    primaryAgentPoolProfiles: [
+      {
+        name: 'agentpool'
+        count: 3
+        vmSize: 'Standard_DS2_v2'
+        osType: 'Linux'
+        mode: 'System'
+        availabilityZones: [
+          1
+        ]
+      }
+    ]
+  }
+}
+
+module createManagedIdentityForDeploymentScript 'br/public:avm/res/managed-identity/user-assigned-identity:0.4.1' = {
+  scope: resourceGroup()
+  name: 'mi-${clusterName}'
+  params: {
+    location: location
+    name: 'mi-${clusterName}'
+  }
+}
+
+// Role assignment for AKS Cluster Contributor access
+resource roleAssignmentAKSClusterContributor 'Microsoft.Authorization/roleAssignments@2022-04-01' = {
+  name: guid(resourceGroup().id, createManagedIdentityForDeploymentScript.name, 'ed7f3fbd-7b88-4dd4-9017-9adb7ce333f8')
+  scope: resourceGroup()
+  properties: {
+    principalId: createManagedIdentityForDeploymentScript.outputs.principalId
+    roleDefinitionId: subscriptionResourceId(
+      'Microsoft.Authorization/roleDefinitions',
+      'ed7f3fbd-7b88-4dd4-9017-9adb7ce333f8'
+    ) // Azure Kubernetes Service Contributor Role
+    principalType: 'ServicePrincipal'
+  }
+}
+
+// Role assignment for Arc Onboarding
+resource roleAssignmentArcOnboarding 'Microsoft.Authorization/roleAssignments@2022-04-01' = {
+  name: guid(resourceGroup().id, createManagedIdentityForDeploymentScript.name, '34e09817-6cbe-4d01-b1a2-e0eac5743d41')
+  scope: resourceGroup()
+  properties: {
+    principalId: createManagedIdentityForDeploymentScript.outputs.principalId
+    roleDefinitionId: subscriptionResourceId(
+      'Microsoft.Authorization/roleDefinitions',
+      '34e09817-6cbe-4d01-b1a2-e0eac5743d41'
+    ) // Kubernetes Cluster - Azure Arc Onboarding Role
+    principalType: 'ServicePrincipal'
+  }
+}
+
+// Role assignment for Resource Group Contributor access
+resource roleAssignmentContributor 'Microsoft.Authorization/roleAssignments@2022-04-01' = {
+  name: guid(resourceGroup().id, createManagedIdentityForDeploymentScript.name, 'b24988ac-6180-42a0-ab88-20f7382dd24c')
+  scope: resourceGroup()
+  properties: {
+    principalId: createManagedIdentityForDeploymentScript.outputs.principalId
+    roleDefinitionId: subscriptionResourceId(
+      'Microsoft.Authorization/roleDefinitions',
+      'b24988ac-6180-42a0-ab88-20f7382dd24c'
+    ) // Contributor Role
+    principalType: 'ServicePrincipal'
+  }
+}
+
+// Reference to the AKS cluster created by the module
+resource aksCluster 'Microsoft.ContainerService/managedClusters@2024-02-01' existing = {
```

Comment:
Does this work upon the first deployment, i.e., when the resource does not yet exist?